kernel/kernel-api/src/main/java/io/delta/kernel/ResolvedTable.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel;
+
+import io.delta.kernel.annotation.Evolving;
+import io.delta.kernel.types.StructType;
+import java.util.List;
+import java.util.Optional;
+
+@Evolving
+public interface ResolvedTable {
+  String getPath();
+
+  long getVersion();
+
+  long getTimestamp();
+
+  List<String> getPartitionColumns();
```

Comment:
not a blocker for merging this but I know we also had discussions about adding `getClusteringColumns` but then we ran into the question of whether we should represent columns as String or `Column` since clustering columns can be nested and cannot be represented just by `String` here

@vkorukanti do you think this is a good time to just switch to `Column` for both (at least for partition columns here)?

Reply:
+1 on switching to a "column" type. BUT, partition columns can only be top level so maybe not needed initially here.

Reply:
cc @vkorukanti any thoughts?

I agree with << maybe not needed initially here. >> -- will leave as is for now until we decide otherwise

Reply:
yeah, Column sounds fine.